backend.go

```diff
@@ -925,12 +925,20 @@ func (backend *Backend) LinearizeMatrix(frags []ValidatedFragment, pieceSize int
 			} else {
 				atomic.AddUint64(&totLen, uint64(outlen))
 			}
+
 			C.freeStrArray(cDataFrags)
 			wg.Done()
 		}(i)
 	}
 	wg.Wait()
 
+	/* Tasks above which call into C.linearize keep a pointer of each
+	   fragment to perform their computation. When all goroutines are in the
+	   ffi call, there is no outstanding reference to frags. This ensure
```

Comment:
Why we don't keep the ref to cDataFrags or frags in ffi call? may be it is impossible?

Reply:
Go doesn't track raw pointer because it is outside of its control. You could compute an arbitrary pointer to a gc object. So when the ffi call is ongoing, there is no longer any references from the PoV of the gc.

Reply:
To elaborate a bit more. Keeping a reference is exactly what KeepAlive does. By using the object as an argument.